docs/guides/auth/oauth.rst

```diff
@@ -197,51 +201,171 @@ pieces of data for an ``auth_token``.
-         res.end(`Error from the auth server: ${text}`);
-         return;
-      }
-
-      const { auth_token } = await codeExchangeResponse.json();
-      res.writeHead(204, {
-         "Set-Cookie": `edgedb-auth-token=${auth_token}; HttpOnly; Path=/; Secure; SameSite=Strict`,
-      });
-      res.end();
+     const requestUrl = getRequestUrl(req);
+
+     const code = requestUrl.searchParams.get("code");
+     if (!code) {
+       const error = requestUrl.searchParams.get("error");
+       res.status = 400;
+       res.end(
+         `OAuth callback is missing 'code'. OAuth provider responded with error: ${error}`
+       );
+       return;
+     }
+
+     const cookies = req.headers.cookie?.split("; ");
+     const verifier = cookies
+       ?.find((cookie) => cookie.startsWith("edgedb-pkce-verifier="))
+       ?.split("=")[1];
+     if (!verifier) {
+       res.status = 400;
+       res.end(
+         `Could not find 'verifier' in the cookie store. Is this the same user agent/browser that started the authorization flow?`
+       );
+       return;
+     }
+
+     const codeExchangeUrl = new URL("token", EDGEDB_AUTH_BASE_URL);
+     codeExchangeUrl.searchParams.set("code", code);
+     codeExchangeUrl.searchParams.set("verifier", verifier);
+     const codeExchangeResponse = await fetch(codeExchangeUrl.href, {
+       method: "GET",
+     });
+
+     if (!codeExchangeResponse.ok) {
+       const text = await codeExchangeResponse.text();
+       res.status = 400;
+       res.end(`Error from the auth server: ${text}`);
+       return;
+     }
+
+     const { auth_token } = await codeExchangeResponse.json();
+     res.writeHead(204, {
+       "Set-Cookie": `edgedb-auth-token=${auth_token}; HttpOnly; Path=/; Secure; SameSite=Strict`,
+     });
+     res.end();
    };
 
 .. lint-on
 
+Creating a User object
+----------------------
+
+Some applications may want to create a custom ``User`` type in their default
+module to attach application-specific information. You can tie this to an
+``ext::auth::Identity`` by using the ``auth_token`` in our
+``ext::auth::client_token`` global and inserting your ``User`` object with a
+link to the ``Identity``.
+
+For this example, we'll assume you have a 1-to-1 relationship between ``User``
+objects and ``ext::auth::Identity`` objects, but in your own application, you
+may decide to have a 1-to-many relationship.
+
+So, given this ``User`` type:
+
+.. code-block:: sdl
+
+   type User {
+       email: str;
+       name: str;
+
+       required identity: ext::auth::Identity {
+           constraint exclusive;
+       };
+   }
+
+You can update the callback function like this to create a new ``User`` object
+when the callback succeeds and the ``isSignUp`` query parameter is set:
+
+.. lint-off
+
+.. code-block:: javascript-diff
+
+     const { auth_token } = await codeExchangeResponse.json();
+   +
+   + const isSignUp = requestUrl.searchParams.get("isSignUp");
+   + if (isSignUp === "true") {
```

Comment:
Should we call out in some way that this depends on having this search parameter set up in `handleAuthorize`, maybe in a note or even in a comment in the code? My concern is that someone followed this example previously and doesn't have that new code to set the param which is added in this PR, sees only this addition, tries to implement, and fails because `isSignup` hasn't been set.

Reply:
That's a good callout, but it might be a little noisy for those who are following along from the beginning. I guess this is a drawback to not having this as like a fully fleshed out example in git to help track changes. Maybe in the proceeding paragraph we can remind the reader that we set up a separate `redirect_to_on_signup` value in the `handleAuthorize` handler?

Reply:
@raddevon ⚡ [`f0153c1` (#7354)](https://github.com/edgedb/edgedb/pull/7354/commits/f0153c1f1adab94945343890e302650079680c40)

what say you?